### PR TITLE
UAF-69 UAF-3596 1099 Reporting Step 9 - Annual Batch

### PR DESCRIPTION
**Pull request by mcalopez**
Files created
1. BatchPayeeFormStep.java used to call generateBatchPayeeForms() on TaxReporting1099ServiceImpl.java
2. ElectronicFilingStep.java

Files Modified
2. TaxReporting1099ServiceImpl.java was modified to add the generateBatchPayeeForms() method.
	* generateBatchPayeeForms()
	* generateBatchPayeeForms(int year, List<Payee> payeeList)
3. TaxHelper.java was modified to add a method that will help create a pdf file
	* getTaxYearFolder(int year, ConfigurationService configurationService, org.apache.log4j.Logger logger)

4. spring-purap.xml was modified to add batchPayeeFormJob and step BatchPayeeFormStep beans. Also, ElectronicFilingStep beans

### CODE REVIEW

File: kfs-purap/src/main/java/edu/arizona/kfs/module/purap/TaxHelper.java [hunk 1]
```diff
@@ -116,5 +120,19 @@ public static boolean isValidIncomeTypeDocumentClass(Class clazz) {
...
 	}
+	
+	public static File getTaxYearFolder(int year, ConfigurationService configurationService, org.apache.log4j.Logger logger) {
+			
+		File retval = new File(configurationService.getPropertyValueAsString(KFSConstants.STAGING_DIRECTORY_KEY) + "/tax/" + year);
+		
+		if (retval.exists() && retval.isDirectory()) {
+			logger.info("tax folder " + retval.getAbsolutePath() + " already exists.");
+		}
+		else if (!retval.mkdirs()) {
```
Comment (ghost) [inline kfs-purap/src/main/java/edu/arizona/kfs/module/purap/TaxHelper.java:131]: KFS 3.0 code is:
` } else if (!retval.mkdirs()) {
            throw new RuntimeException("Could not create folder at [" + retval.getAbsolutePath() + "].");
        } else {
            logger.info("tax folder " + retval.getAbsolutePath() + " created.");
        }`
Why the change? Why are we logging that the folder was created when mkdirs() returned an error condition?

**Review by ghost — COMMENTED** [review]
Initial review complete. Please see my comment.

File: kfs-purap/src/main/java/edu/arizona/kfs/module/purap/TaxHelper.java (hunk 1 above)
Comment (mcalopez) [reply kfs-purap/src/main/java/edu/arizona/kfs/module/purap/TaxHelper.java:131]: My bad, I didn't realize I skipped this part.

**Comment by ghost** [comment]
Looks good! Waiting to merge until current release is complete.
